kotlin/vector-tiled-layer-custom-style/src/main/java/com/esri/arcgisruntime/sample/vectortiledlayercustomstyle/MainActivity.kt

```diff
@@ -0,0 +1,293 @@
+                val viewpoint = Viewpoint(Point(-100.01766, 37.76528, SpatialReferences.getWgs84()), 100000.0)
+                setMap(layer, viewpoint)
+            }
+        }
+    }
+
+    /**
+     * Set the map using the layer and the viewpoint.
+     */
+    private fun setMap(layer: ArcGISVectorTiledLayer, viewpoint: Viewpoint){
+        //Reset the map to release resources
+        mapView.map = null
+        //mapView.map.basemap = null
+
+        // Assign a new map created from the base layer.
+        val basemap = Basemap(layer.copy())
+        val map = ArcGISMap(basemap)
+        mapView.map = map
+        //Set viewpoint without animation.
+        mapView.setViewpoint(viewpoint)
+    }
+
+    /**
+     * Sets the adapter and listens for a item selection to update the
+     * MapView with the selected layer.
+     */
+    private fun setUpSpinner() {
+        val customDropDownAdapter = CustomDropDownAdapter(this)
+        spinner.adapter = customDropDownAdapter
+        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+
+                // Sets the selected itemID to either the Online/Custom ID.
+                currentItemID = when (position) {
+                    4 -> {
+                        // Custom style 1 - Dodge City OSM - Light
+                        offlineItemIds[0]
+                    }
+                    5 -> {
+                        // Custom style 2 - Dodge City OSM - Dark
+                        offlineItemIds[1]
+                    }
+                    else -> {
+                        // Else use the online vector styles
+                        onlineItemIds[position]
+                    }
+                }
+                showSelectedItem(currentItemID)
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                //Keeps the current selected vector style.
+            }
+        }
+    }
+
+    //[DocRef: Name=Pause and resume-Android, Category=Get started, Topic=Develop your first map app with Kotlin]
+    override fun onPause() {
+        mapView.pause()
+        super.onPause()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mapView.resume()
+    }
+
+    override fun onDestroy() {
+        mapView.dispose()
+        super.onDestroy()
+    }
+
+    //[DocRef: END]
+
+    class CustomDropDownAdapter(private val context: Context) : BaseAdapter() {
+
+        // The names of the each Vector Tiled Layer
+        private val styleNames = arrayOf("Default", "Style 1", "Style 2", "Style 3", "Offline custom style: Light", "Offline custom style: Dark")
+
+        // The drawable XML file names for the associated style name
+        private val styleDrawableNames = arrayOf("default_color","style1_color","style2_color","style3_color","custom1_color","custom2_color")
```

Comment:
The linter might ask you to put these in the strings.xml file, which probably isn't a bad idea in this case